src/mapeo-project.js

```diff
@@ -749,6 +761,410 @@ export class MapeoProject extends TypedEmitter {
+            coordinates,
+          }
+        : null
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: observation,
+            geometry,
+          })
+      )
+    }
+  }
+
+  /**
+   * @param {Iterable<Track>} tracks
+   * @param {Object} options
+   * @param {Set<string>} [options.seenObservations=new Set()]
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @param {string} [options.lang]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportTracks(
+    tracks,
+    { lang, seenObservations = new Set(), seenAttachments = new Map() } = {}
+  ) {
+    let first = true
+    for (const track of tracks) {
+      const { observationRefs } = track
+
+      const observations = await Promise.all(
+        observationRefs.map(({ docId }) =>
+          this.#dataTypes.observation.getByDocId(docId, { lang })
+        )
+      )
+
+      const coordinates = track.locations.map(
+        ({ coords: { longitude, latitude, altitude } }) => [
+          longitude,
+          latitude,
+          altitude,
+        ]
+      )
+      const comma = first ? '' : ','
+      first = false
+      yield b4a.from(
+        `${comma}\n      ` +
+          JSON.stringify({
+            type: 'Feature',
+            properties: track,
+            geometry: {
+              type: 'LineString',
+              coordinates,
+            },
+          }) +
+          '\n'
+      )
+
+      let firstObs = true
+      for await (const chunk of this.#exportObservations(observations, {
+        seenObservations,
+        seenAttachments,
+      })) {
+        if (firstObs) {
+          yield b4a.from(',')
+          firstObs = false
+        }
+        yield chunk
+      }
+    }
+  }
+
+  /**
+   * @param {Object} [options={}]
+   * @param {boolean} [options.observations=true] Whether observations should be exported
+   * @param {boolean} [options.tracks=true] Whether all tracks and their observations should be exported
+   * @param {SeenAttachments} [options.seenAttachments]
+   * @param {string} [options.lang]
+   * @returns {AsyncIterable<Buffer | Uint8Array>}
+   */
+  async *#exportGeoJSONIterator({
+    observations = true,
+    tracks = true,
+    lang,
+    seenAttachments = new Map(),
+  } = {}) {
+    yield b4a.from(`{
+  "type": "FeatureCollection",
+  "features": [
+`)
+
+    const seenObservations = new Set()
+
+    let hadTracks = false
+    if (tracks) {
+      const rows = await this.#dataTypes.track.getMany({ lang })
+      for await (const chunk of this.#exportTracks(rows, {
+        lang,
+        seenObservations,
+        seenAttachments,
+      })) {
+        hadTracks = true
+        yield chunk
+      }
+    }
+
+    if (observations) {
+      if (hadTracks) {
+        yield b4a.from(',')
+      }
+      const rows = await this.#dataTypes.observation.getMany({ lang })
```

Comment:
Can we put a pin in a possible follow up to expose an iterable getMany from bettersqlite3, to avoid needing to load all data into memory as part of the export.

Reply:
Yeah my original implementation was to do a stream of the observations but I found out that sqlite doesn't support iterating over a query. https://orm.drizzle.team/docs/select#iterator

It's whack that streaming isn't a first class feature of all the adapters 😅

We could make one by using `SKIP`/`LIMIT` to do pagination. Thoughts on page size? I was thing 512 or 1024.

Reply:
Tracked https://github.com/digidem/comapeo-core/issues/1038

Reply:
Ah interesting. It's supported by bettersqlite3, but yeah it does seem that drizzle doesn't support this yet. I don't think this is an urgent issue, since we load everything into memory for reads anyway. Let's punt this and address it when we add pagination to reads in the API, and then we can use that pagination for export too.